### PR TITLE
Add password complexity checks for openhab-cloud account

### LINKED ISSUE #75: Improper password strength controls
Description:
A strong password is a key requirement to protect against password guessing attacks. The application does not implement a strong password policy, as users can create an account with a password of one character.

Steps to reproduce (A):
1. Authenticate
2. Go to <baseurl>/account
3. Enter a new password of one character, e.g. 'a'.
4. Save
5. Logout
6. Authenticate with the same user and the new password. Observe that this worked.

More information: https://www.owasp.org/index.php/Authentication_Cheat_Sheet#Implement_Proper_Password_Strength_Controls

### PR DESCRIPTION
**Pull request by FlorianSW**
All places, where a password can be set will now go through a new class,
UserPassword, which will check the password complexity before setting the
password to the user account. It also provides a check method to check the
complexity requirements before a user account will be created. If the
complexity rules aren't met, the save of the password will be denied and an
error message with the complexity rules is shown.

Fixes #75

### CODE REVIEW

**Comment by marziman** [comment]
@FlorianSW,
thanks! Going to review this as soon as possible
BR Mehmet

**Comment by digitaldan** [comment]
Hi @FlorianSW this works well when changing a password, but we also let users choose a password when they create an account and I don't think this is getting used there?

File: routes/account.js [hunk 1]
```diff
@@ -343,6 +354,12 @@ exports.registerpost = function(req, res) {
                       res.render('login', { title: "Login / Sign up", user: req.user,
                           errormessages:req.flash('error'), infomessages:req.flash('info') });
                     } else {
+                        if (!UserPassword.isComplexEnough(req.body.password)) {
```
Comment (FlorianSW) [inline routes/account.js]: This should be the place where a user selected a password during registration, isn't it?

**Comment by FlorianSW** [comment]
Hi @digitaldan :) Thanks for your review! It's totally possible, that I missed a place, however, see my comment, where I think the registration password should be handled? Do I miss another place, too? Thanks for your help :)

**Comment by digitaldan** [comment]
It looks like it, but for some reason it let me create an account with 1234 as my password, let me try again and make sure my password manager didn't autofill something in right before i hit submit (lastpass does that sometimes).

**Comment by FlorianSW** [comment]
Interesting, I checked this, too (I remember that, because I already wrote git push and remembered, that I forgot to change behaviour of the register form and I had to implement that before I pushed :P). I'll check that again, just to be sure :)

**Comment by digitaldan** [comment]
I just tried it again and now it works, I'm going to blame lastpass :-)

**Comment by FlorianSW** [comment]
I tested it again, too, and it's working great *yeah*

Thanks for merging!

**Comment by marziman** [comment]
@FlorianSW maybe I am stupid, but can it be that it will also force to change passed for existing users with no good pw complexity. I have some local test accounts and they don't work anymore. This is not bad since it forces users to have better pw complexity.
This would also lead all users to change passwords.

**Comment by marziman** [comment]
@FlorianSW looks good!
